docs/connecting-data/connecting-to-apis-and-external-data/affinity-api.md

```diff
@@ -4,4 +4,148 @@ description: How to Connect DataDistillr to the Affinity API
 
 # Affinity API
 
-Details coming soon!
\ No newline at end of file
+### Cost
+
+You can [contact](https://www.affinity.co/request-demo?utm_campaign=Capterra&utm_source=capterra&utm_medium=ppc){target=_blank} Affinity for a free demo and pricing.
+
+### Rate Limits
+
+Your account plan tier will limit the number of requests each API key can make per month. Current rate limits by plan tier are:
+Proffesional tier gets 40,000 calls; Premium tier gets 100,000 calls; Enterprise tier gets unlimited calls.
+
+All API requests will be halted at 450K per user, per day. Your daily limit will reset the next day at 12AM (midnight) Pacific Time.
+
+## How to Connect DataDistillr to Affinity
+To set up a data source connect for Affinity, you will need to have:
+
+- A [unique name](#name) for your data source connection to be used in queries
+- An [API token](#api-key) generated through your Affinity account
+
+
+### Data Source Form
+To locate the Affinity form, follow the steps in [Connecting Your Data to DataDistillr](../../). When you get to the window to choose the data source type, select API as shown below.
+
+<figure markdown>
+  ![Data Source Wizard][image-9]{ width="100%" }
+</figure>
+
+On the API screen, select Affinity from the list of API forms.
+
+<figure markdown>
+  ![List of APIs][image-3]{ width="100%" }
+</figure>
+
+The following form will appear. Instructions can be found below on how to find the information required to fill each field on the Affinity API form.
+
+Once you have filled out all the fields, press the green 'Save' button, and your API will be connected!
+
+
+<figure markdown>
+  ![Affinity Form][image-5]{ width="100%" }
+</figure>
+
+
+### Name
+Enter any name that will help you recognize this data source within your query window.
+
+!!! info "Acceptable characters include"
+
+    - lowercase alphanumeric characters
+    - underscores
+
+
+### API Key
+The API key is generated within your account's settings page. Head over to the [Affinity docs](https://support.affinity.co/hc/en-us/articles/360032633992-How-to-obtain-your-API-Key){target=_blank} to learn how to generate your API Key. Once created, copy the key and enter it in the Affinity form.
+
+## Endpoints
+The table below shows a list of endpoints available to connect within the DataDistillr application. If you need to connect to any endpoints not listed in the table below, please use the [Custom API](../../) Form.
+
+| Endpoint       | Required  | URL Parameters           | Optional                                                        | Description                                                                                                         |
```

Comment:
Change the order of URL parameters and Optional in the table - feels awkward to split up Required and Optional